pycobertura/utils.py

```diff
@@ -229,3 +231,29 @@ def hunkify_lines(lines, context=3):
 
 def get_dir_from_file_path(file_path):
     return os.path.dirname(file_path) or "."
+
+
+def get_non_empty_non_commented_lines_from_file_in_ascii(file_path, comment_character):
+    with open(file_path, "rb") as f:  # read in binary (more secure)
+        result = [
+            line.decode("ascii").strip()
+            for line in f.readlines()
+            if not line.decode("ascii").startswith(comment_character)
+        ]
+        return [res for res in result if res != ""]
+
+
+def get_filenames_that_do_not_match_regex(
```

Comment:
Could we use a tool like this one? https://github.com/mherrmann/gitignore_parser – would it have the same effect?

Reply:
The effect would be similar (it though does only handle files), but this library did not have any changes since August 2020 and my implementation seems to use more Python in-built functions. Also the former does not have a stable semantic version (0.0.8). In priniciple, best practice is to only include well-maintained libraries and such that have a major version >= 1.

Reply:
@aconrad Maybe this library could be used: https://github.com/bitranox/igittigitt. And also tell me if you have any other things still needed here to get into a good state to merge - thanks :)